rust/connlib/tunnel/src/client.rs

```diff
@@ -812,6 +816,16 @@ impl ClientState {
             .map(|(ip, _)| ip)
             .chain(iter::once(IPV4_RESOURCES.into()))
             .chain(iter::once(IPV6_RESOURCES.into()))
+            .chain(
```

Comment:
I don't like `map` here because I don't think expresses what's happening to well, since we want to ignore the content of the `Option`.

This is equivalent to the `map`  but `.is_some().then_some(...)` seems a bit more expressive.